### ISSUE TITLE
Navigation macro does not recognize explicit false for sub_path

### ISSUE THREAD

**Issue by SirNate0** [gollum/gollum#2023]
**Describe the bug**
The [Navigation macro](https://github.com/gollum/gollum/wiki/Standard-Macros#navigation)'s final parameter `full_path` is false by default. However, specifying it as false explicitly gives the same result as setting it to true. 

**Expected behavior**
`<<Navigation(Title, subdir, false)>>` should behave the same as `<<Navigation(Title, subdir)>>` (which does have the correct behavior of not including the full path), rather than behaving like `<<Navigation(Title, subdir, true)>>` (which shows `subdir` in the path).

**Environment Info**

```
Gollum 5.3.2
Running on: x86_64-linux-gnu with Ruby version 3.0.2
Using:
rugged 1.7.1
gollum-rugged_adapter 2.1.0
gollum-lib 5.2.4
Markdown rendering gem: kramdown
Other renderering gems:
none
```

**Comment by benjaminwil** [comment]
Thanks for reporting!

If you (or anyone reading this) would provide step-by-step instructions for reproduction and describe the output vs. the expected output, that would help developers easily reproduce this issue and make it more straightforward to start fixing the issue.

I haven't tried reproducing this myself, yet, but I would assume we need at least one page in a subdirectory to reproduce?

**Comment by SirNate0** [comment]
I would imagine so, otherwise you would have nothing printed and would be unable to see whether the subdirectory is printed or not. This works for me:

* Home.md
    ```md

    <<Navigation("With False", "Test", false)>>
    
    <<Navigation("With True", "Test", true)>>
    
    <<Navigation("With Nothing", "Test")>>
    ```
* Test/dA/A.md
    ```md
    Some text
    ````
* Test/dA/B.md
    ```md
    Other text
    ````
* Test/dB/B.md
    ```md
    Other text
    ````

**Comment by dometto** [comment]
I tried but I can't reproduce this on gollum 6.0. From the file `Bilbo-Baggins.md` in our standard test repo (`test/examples/lotr.git` in the main gollum repo):

<img width="773" alt="Screenshot 2024-07-23 at 15 55 34" src="https://github.com/user-attachments/assets/216e5338-199f-4318-abce-2d17da9e313a">

I'm assuming this is a bug that was (accidentally?) fixed in version 6.0 as @SirNate0 was using 5.2.4. 

@SirNate0: please try updating to the newest release of gollum, but do let us know if the issue crops up again.
